### DISCUSSION TITLE
"pages" argument in pdfplumber.open

### DISCUSSION THREAD

**Discussion by vinaychuri** [jsvine/pdfplumber#697] [Q&A] [answered]
The function _pdfplumber.open("xxxx.pdf")_ has a **_pages_** argument but I am unable to set it correctly.

My pdf file is 70 pages and I want to extract only the first 5 pages. In another example i want to only extract only certain pages.
>>>pdf = pdfplumber.open("xxxx.pdf",pages="1-5")
>>>pdf = pdfplumber.open("xxxx.pdf",pages="1,5")
>>>pdf = pdfplumber.open("xxxx.pdf",pages="1:5")

All the above options give the following error.
>>>pdf.pages
#TypeError: 'in <string>' requires string as left operand, not int

I can iterate the "pdf.pages" and extract the required tables but would be great if **pdfplumber.open** could use the _**pages**_ argument.
Thanks

**Comment by jsvine** [comment, marked as answer]
Hi, `pages` expects a list or `range(...)` of page numbers, e.g., `pages=[1, 2, 3, 4, 5]` or `pages=range(1, 6)`.
